syntax = "proto3";

enum DataPath {
    AF_PACKET = 0;
    AF_XDP_ZC = 1;
}

enum TxSelection{
    EST = 0;
    STRICT_PRIO = 1;
}


message InitRequest {
	string interface = 1;
	bool hints_available = 2;
	TxSelection hints_tx_selection = 3;
	bool hints_tx_selection_offload = 4;
	DataPath hints_data_path = 5;
	bool hints_preemption = 6;
	bool hints_launch_time_control = 7;
}


message InitResponse {
	bool ok = 1;
}


message StreamQosRequest {
	string interface = 1;
	uint32 period = 2;
	uint32 size = 3;
	string dmac = 4;
	uint32 vid = 5;
	uint32 pcp = 6;
	uint32 txmin = 7;
	uint32 txmax = 8;
	bool   setup_socket = 9;
	uint32 basetime = 10;
	bool talker = 11;
	string maddress = 12;
}


message StreamQosResponse {
	bool    ok = 1;
	string  vlan_interface = 2;
	uint32  socket_priority = 3;
}


message DetdMessage {
	oneof msg {
	    InitRequest       init_request = 1;
	    InitResponse      init_response = 2;
	    StreamQosRequest  stream_qos_request = 3;
	    StreamQosResponse stream_qos_response = 4;
	}
}
